### PR TITLE
Add VotingEscrowRemapper interface

### PR DESCRIPTION
**Pull request by nventuro**
# Description

This extracts the interface into its own file, and restores the `getUserPoint` function.

## Type of change

- [ ] Bug fix 
- [x] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [x] The diff is legible and has no extraneous changes
- [x] Complex code has been commented, including external interfaces
- [x] Tests are included for all code paths
- [x] The base branch is either `master`, or there's a description of how to merge

### CODE REVIEW

File: pkg/interfaces/contracts/liquidity-mining/IVotingEscrowRemapper.sol [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    /**
+     * @notice Returns the current total supply of veBAL as a Point.
+     * @dev We return the total supply as a Point to allow extrapolating this into the future. Note that this
+     * extrapolation will become invalid when crossing weeks, since we're not taking into account veBAL locks that
+     * expire then.
+     */
+    function getTotalSupplyPoint() external view returns (IVotingEscrow.Point memory);
+
+    /**
+     * @notice Returns a user's current veBAL balance as a Point. Note that we don't apply any remappings in this query.
```
Comment (jubeira) [inline pkg/interfaces/contracts/liquidity-mining/IVotingEscrowRemapper.sol:53]:
```suggestion
     * @notice Returns a user's current veBAL balance as a Point. Note that we don't apply any remappings in this query.
     * The returned value is taken directly from the voting escrow.
```

File: pkg/interfaces/contracts/liquidity-mining/IVotingEscrowRemapper.sol [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+     * @notice Returns the remote user corresponding to an address on the local chain.
+     * @dev Returns `address(0)` if the remapping does not exist for the given local user.
+     * @param localUser - Address of the user on the local chain which are querying the remote address for.
+     * @param chainId - The chain ID of the network which the remote user is on.
+     */
+    function getRemoteUser(address localUser, uint256 chainId) external view returns (address);
+
+    function getRemappingManager(address localUser) external view returns (address);
+
+    function setOmniVotingEscrow(IOmniVotingEscrow omniVotingEscrow) external;
```
Comment (jubeira) [inline pkg/interfaces/contracts/liquidity-mining/IVotingEscrowRemapper.sol:82]: Let's add docs now that we are at it.

File: pkg/interfaces/contracts/liquidity-mining/IVotingEscrowRemapper.sol [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+     * @param remoteUser - The address to receive `localUser`'s balance of veBAL on the remote chain.
+     * @param chainId - The chain id of the remote chain on which `remoteUser` resides.
+     */
+    function setNetworkRemapping(
+        address localUser,
+        address remoteUser,
+        uint256 chainId
+    ) external payable;
+
+    function setNetworkRemappingManager(address localUser, address delegate) external;
```
Comment (jubeira) [inline pkg/interfaces/contracts/liquidity-mining/IVotingEscrowRemapper.sol:100]:
```suggestion
    /**
     * @notice Sets an address to manage the mapping for a given local user on its behalf.
     * @dev This is intended to handle contracts which cannot interact with this contract directly.
     * @param localUser - The address of a contract allowlisted on the `SmartWalletChecker`.
     * @param delegate - The address which is allowed to manage remote users to be linked to `localUser`.
     */
    function setNetworkRemappingManager(address localUser, address delegate) external;
```

File: pkg/liquidity-mining/contracts/VotingEscrowRemapper.sol [hunk 4]
```diff
@@ -42,93 +40,58 @@ contract VotingEscrowRemapper is SingletonAuthentication, ReentrancyGuard {
...
-
     constructor(IVotingEscrow votingEscrow, IVault vault) SingletonAuthentication(vault) {
         _votingEscrow = votingEscrow;
     }
 
-    /**
-     * @notice Returns Voting Escrow contract address.
-     */
-    function getVotingEscrow() public view returns (IVotingEscrow) {
+    function getVotingEscrow() public view override returns (IVotingEscrow) {
```
Comment (jubeira) [inline pkg/liquidity-mining/contracts/VotingEscrowRemapper.sol:47]: Here and elsewhere: we can use `@inheritdoc IVotingEscrowRemapper` now.

File: pkg/liquidity-mining/test/VotingEscrowRemapper.test.ts [hunk 5]
```diff
@@ -560,6 +560,39 @@ describe('VotingEscrowRemapper', function () {
...
+
+    it('is unaffected by remappings', async () => {
+      await smartWalletChecker.connect(admin).allowlistAddress(local.address);
+      await vault.grantPermissionGlobally(await actionId(remapper, 'setOmniVotingEscrow'), admin);
+      await remapper.connect(admin).setOmniVotingEscrow(omniVotingEscrow.address);
+
+      await remapper.connect(local).setNetworkRemapping(local.address, remote, 1);
+      expect(await remapper.getRemoteUser(local.address, 1)).to.equal(remote);
+
+      expect(await remapper.getUserPoint(local.address)).to.be.deep.eq(Object.values(point));
```
Comment (jubeira) [inline pkg/liquidity-mining/test/VotingEscrowRemapper.test.ts:592]: let's also add `getUserPoint(remote)` and verify that it returns 0.

**Review by jubeira — COMMENTED** [review]
Initial pass before takeover.

File: pkg/interfaces/contracts/liquidity-mining/IVotingEscrowRemapper.sol [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+    event RemoteAddressMappingCleared(address indexed remoteUser, uint256 indexed chainId);
+    event AddressDelegateUpdated(address indexed localUser, address indexed delegate);
+
+    /**
+     * @notice Returns Voting Escrow contract address.
+     */
+    function getVotingEscrow() external view returns (IVotingEscrow);
+
+    /**
+     * @notice Returns Omni Voting Escrow contract address.
```
Comment (EndymionJkb) [inline pkg/interfaces/contracts/liquidity-mining/IVotingEscrowRemapper.sol:43]: Maybe say more about what this is? (Looks like it's specific to Layer 0.)

File: pkg/interfaces/contracts/liquidity-mining/IVotingEscrowRemapper.sol [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+     */
+    function getOmniVotingEscrow() external view returns (IOmniVotingEscrow);
+
+    /**
+     * @notice Returns the current total supply of veBAL as a Point.
+     * @dev We return the total supply as a Point to allow extrapolating this into the future. Note that this
```
Comment (EndymionJkb) [inline pkg/interfaces/contracts/liquidity-mining/IVotingEscrowRemapper.sol:49]: Maybe link to IVotingEscrow where Point is defined? Though I guess the interface name is already given in the return value, so maybe that's good enough.

File: pkg/interfaces/contracts/liquidity-mining/IVotingEscrowRemapper.sol [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+    /**
+     * @notice Get timestamp when `user`'s lock finishes.
+     * @dev The returned value is taken directly from the voting escrow.
+     */
+    function getLockedEnd(address user) external view returns (uint256);
+
+    /**
+     * @notice Returns the local user corresponding to an address on a remote chain.
+     * @dev Returns `address(0)` if the remapping does not exist for the given remote user.
+     * @param remoteUser - Address of the user on the remote chain which are querying the local address for.
```
Comment (EndymionJkb) [inline pkg/interfaces/contracts/liquidity-mining/IVotingEscrowRemapper.sol:71]:
```suggestion
     * @param remoteUser - Address of the user on the remote chain corresponding to the local address.
```

File: pkg/interfaces/contracts/liquidity-mining/IVotingEscrowRemapper.sol [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+     * @param chainId - The chain ID of the network which this user is on.
+     */
+    function getLocalUser(address remoteUser, uint256 chainId) external view returns (address);
+
+    /**
+     * @notice Returns the remote user corresponding to an address on the local chain.
+     * @dev Returns `address(0)` if the remapping does not exist for the given local user.
+     * @param localUser - Address of the user on the local chain which are querying the remote address for.
```
Comment (EndymionJkb) [inline pkg/interfaces/contracts/liquidity-mining/IVotingEscrowRemapper.sol:79]:
```suggestion
     * @param localUser - Address of the user on the local chain corresponding to the remote address.
```

File: pkg/interfaces/contracts/liquidity-mining/IVotingEscrowRemapper.sol [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+     * @param chainId - The chain ID of the network which the remote user is on.
```
Comment (EndymionJkb) [inline pkg/interfaces/contracts/liquidity-mining/IVotingEscrowRemapper.sol:80]:
```suggestion
     * @param chainId - The chain ID of the remote network.
```

File: pkg/interfaces/contracts/liquidity-mining/IVotingEscrowRemapper.sol [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+     * @notice Get timestamp when `user`'s lock finishes.
+     * @dev The returned value is taken directly from the voting escrow.
+     */
+    function getLockedEnd(address user) external view returns (uint256);
+
+    /**
+     * @notice Returns the local user corresponding to an address on a remote chain.
+     * @dev Returns `address(0)` if the remapping does not exist for the given remote user.
+     * @param remoteUser - Address of the user on the remote chain which are querying the local address for.
+     * @param chainId - The chain ID of the network which this user is on.
```
Comment (EndymionJkb) [inline pkg/interfaces/contracts/liquidity-mining/IVotingEscrowRemapper.sol:72]:
```suggestion
     * @param chainId - The chain ID of the remote network.
```

File: pkg/interfaces/contracts/liquidity-mining/IVotingEscrowRemapper.sol [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+     * @notice Returns the remote user corresponding to an address on the local chain.
+     * @dev Returns `address(0)` if the remapping does not exist for the given local user.
+     * @param localUser - Address of the user on the local chain which are querying the remote address for.
+     * @param chainId - The chain ID of the network which the remote user is on.
+     */
+    function getRemoteUser(address localUser, uint256 chainId) external view returns (address);
+
+    /**
+     * @notice Gets the account that can set a remapping for a given local user.
+     * @param localUser - Address of the user on the local chain which are querying the remapping manager for.
```
Comment (EndymionJkb) [inline pkg/interfaces/contracts/liquidity-mining/IVotingEscrowRemapper.sol:86]:
```suggestion
     * @param localUser - Address of the user on the local chain with a remapping manager.
```

Comment (EndymionJkb) [reply pkg/interfaces/contracts/liquidity-mining/IVotingEscrowRemapper.sol:86]: Say what it does if not found? Does it return the zero?

File: pkg/interfaces/contracts/liquidity-mining/IVotingEscrowRemapper.sol [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+     * @param chainId - The chain id of the remote chain on which `remoteUser` resides.
+     */
+    function setNetworkRemapping(
+        address localUser,
+        address remoteUser,
+        uint256 chainId
+    ) external payable;
+
+    /**
+     * @notice Sets an address to manage the mapping for a given local user on its behalf.
```
Comment (EndymionJkb) [inline pkg/interfaces/contracts/liquidity-mining/IVotingEscrowRemapper.sol:115]:
```suggestion
     * @notice Sets an address to manage the mapping for a given local user on their behalf.
```

File: pkg/interfaces/contracts/liquidity-mining/IVotingEscrowRemapper.sol [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+     * @param localUser - The address of a contract allowlisted on the `SmartWalletChecker`.
+     * @param delegate - The address which is allowed to manage remote users to be linked to `localUser`.
+     */
+    function setNetworkRemappingManager(address localUser, address delegate) external;
+
+    /**
+     * @notice Clears a local user's mapping for a particular network.
+     * @dev This is intended to discourage and also allow recovery from griefing attacks.
+     * If griefing occurs then the griefer can be removed from Smart Wallet Checker and have their remappings erased.
+     * The local user can always clear its own mapping, regardless the state of the Smart Wallet Checker.
```
Comment (EndymionJkb) [inline pkg/interfaces/contracts/liquidity-mining/IVotingEscrowRemapper.sol:126]:
```suggestion
     * The local user can always clear their own mapping, regardless the state of the Smart Wallet Checker.
```

**Review by EndymionJkb — APPROVED** [review]
Just some doc nits - otherwise looks like just interface extraction. The only code change is adding in `getUserPoint`, which you mention in the description and include a test for.

**Pull request merged**